tests/unit/commands/local/lib/test_cfn_api_provider.py

```diff
@@ -1186,6 +1186,86 @@ def test_empty_integration_array(self):
+        assert mock_collector.add_routes.call_args.args[1][0].authorizer_name is None
+
+    @patch("samcli.lib.providers.cfn_api_provider.CfnApiProvider._extract_cfn_gateway_v2_authorizer")
+    @patch("samcli.lib.providers.cfn_api_provider.CfnApiProvider._extract_cloud_formation_authorizer")
+    def test_extract_resources_skips_authorizer_resource(self, mock_extract_cfn_gateway_v2_authorizer: Mock, mock_extract_cloud_formation_authorizer: Mock):
+        template = {
+            "Resources": {
+                "AuthorizerV2": {
+                    "Type": "AWS::ApiGatewayV2::Authorizer",
+                    "Properties": {
+                        "ApiId": "DummyId",
+                        "AuthorizerType": "REQUEST",
+                        "Name": "my-authorizer-v2"
+                    }
+                },
+                "Authorizer": {
+                    "Type": "AWS::ApiGateway::Authorizer",
+                    "Properties": {
+                        "Name": "my-authorizer",
+                        "RestApiId": "DummyId",
+                        "Type": "TOKEN"
+                    }
+                }
+            }
+        }
+        provider = CfnApiProvider()
+        mock_collector = Mock()
+
+        provider.extract_resources(stacks=make_mock_stacks_from_template(template), collector=mock_collector, disable_authorizer=True)
+        mock_extract_cfn_gateway_v2_authorizer.assert_not_called() # This should not be called if disable_authorizers == True
+        mock_extract_cloud_formation_authorizer.assert_not_called()
+        provider.extract_resources(stacks=make_mock_stacks_from_template(template), collector=mock_collector, disable_authorizer=False)
+        mock_extract_cfn_gateway_v2_authorizer.assert_called_once() # Testing inverse
+        mock_extract_cloud_formation_authorizer.assert_called_once()
+    
+    def test_extract_api_gateway_method(self):
+        template = {
+            "MyMethod": {
+                "Type": "AWS::ApiGateway::Method",
+                "Properties": {
+                    "HttpMethod": "GET",
+                    "ResourceId": "DummyId",
+                    "RestApiId": "DummyId",
+                    "AuthorizerId": "AuthorizerId"
+                }
+            }
+        }
+        provider = CfnApiProvider()
+        mock_collector = Mock()
+        mock_collector.add_routes = Mock()
+
+        provider._extract_cloud_formation_method(stack_path="", resources={}, logical_id="MyMethod", method_resource=template["MyMethod"], collector=mock_collector, disable_authorizer=True)
+        assert mock_collector.add_routes.call_args.args[1][0].authorizer_name is None
+        provider._extract_cloud_formation_method(stack_path="", resources={}, logical_id="MyMethod", method_resource=template["MyMethod"], collector=mock_collector, disable_authorizer=False)
+        assert mock_collector.add_routes.call_args.args[1][0].authorizer_name is "AuthorizerId"
+
```

Comment:
Similar comment as above (using `unittest`'s assertions). Could we also parameterize this test case? There are some examples on how to parameterize tests here: https://github.com/aws/aws-sam-cli/blob/09f5fdc3c299d68197853a8dca5e12b0044d60b7/tests/unit/local/apigw/test_lambda_authorizer.py

Reply:
Doesn't seem like there is a clean way of asserting something was called with a partial matcher. I would have to create a Route object that matches exactly - thinking it's better to extract and assert the auth name than adding noise to the test. Thumbs this up if you still want me to create the route obj

Reply:
Since the only part of that assertion that is changing is the name (which depends on if the disable flag was passed), this could be parameterized in a way where the test cases could be:
- `(True, None)`
- `(False, "AuthorizerId)`

It'll be a bit more setup code to get it to this state, but would potentially be easier to modify this test in the future if there were other checks that we wanted to make/behaviours that we wanted to updated.